content/2022-05-19-leave-the-yaml-at-home.md

```diff
@@ -0,0 +1,185 @@
+      described
+      [here](https://github.com/brigadecore/brigade-github-gateway/blob/main/docs/EVENT_REFERENCE.md).)
+
+    * The `workerTemplate` portion of the file describes the worker that should
+      handle the events. In this simple case, it only uses the `git.cloneURL`
+      field to indicate where project source -- including the script we'd like
+      to run -- can be obtained.
+
+This is all the YAML you write! (And, actually, most of this can be generated
+for you with the `brig init` command!)
+
+For good measure, let's mention that project definitions such as this one can be
+submitted to your Brigade API server like so:
+
+```shell
+$ brig project create --file project.yaml
+```
+
+With this very small bit of YAML out of the way, _everything else you do is with
+JavaScript or TypeScript_ -- both of which are enormously popular with developers
+and are, we think, superior methods of modeling complex workflows (as compared
+to yards and yards of YAML).
+
+Your script can be as simple as:
+
+```javascript
+console.log("Hello, World!")
+```
+
+Or could mux events, dispatching them to the correct handler function:
+
+```javascript
+const { events } = require("@brigadecore/brigadier");
+
+events.on("brigade.sh/github", "ci:pipeline_requested", async event => {
+  // Script your CI workflow here
+});
+
+events.on("brigade.sh/github", "cd:pipeline_requested", async event => {
+  // Script your CD workflow here
+});
+
+events.process();
+```
+
+You can describe your workflows as arbitrarily complex graphs of __jobs__ --
+where, behind the scenes, each job is an assembly of one or more containers
+running in its own Kubernetes pod.
+
+Here's a simplified excerpt from Brigade's own `brigade.ts` script:
+
+```typescript
+import { events, Job } from "@brigadecore/brigadier"
+
+const goImg = "brigadecore/go-tools:v0.9.0"
+const srcPath = "/src"
+// ...
+
+events.on("brigade.sh/github", "ci:pipeline_requested", async event => {
+  const testJob = new Job("test", goImg, event)
+  testJob.primaryContainer.command = ["make"]
+  testJob.primaryContainer.arguments = ["test-unit"]
+  testJob.primaryContainer.sourceMountPath = srcPath
+  testJob.primaryContainer.workingDirectory = srcPath
+
+  const lintJob = new Job("lint", goImg, event)
+  lintJob.primaryContainer.command = ["make"]
+  lintJob.primaryContainer.arguments = ["lint"]
+  lintJob.primaryContainer.sourceMountPath = srcPath
+  lintJob.primaryContainer.workingDirectory = srcPath
+
+  // ...
+
+  await Job.concurrent(
+    testJob,
+    lintJob,
+    // ...
+  ).run()
+})
+
+// ...
+
+events.process();
+```
+
+In the excerpt above, we respond to `ci:pipeline_requested` events by executing
+`make test-unit` and `make lint` concurrently (subject to scheduling
+constraints). Each executes in its own container, in its own pod. Those
+containers are both based on a `brigadecore/go-tools:v0.9.0` image that's
+pre-loaded with all of our team's most frequently used Go-based tools.
```

Comment:
just in case people don't know what `Go` refers to:

```suggestion
pre-loaded with all of our team's most frequently used [Go](https://golang.org)-based tools.
```